evmap/src/write.rs

```diff
@@ -588,6 +527,31 @@ where
             unsafe { Box::from_raw(Box::into_raw(self) as *mut _ as *mut _) };
         drop(inner);
     }
+
+    fn sync_with(&mut self, first: &Self) {
+        let inner: &mut Inner<K, V, M, S, crate::aliasing::DoDrop> =
+            unsafe { &mut *(self as *mut _ as *mut _) };
+        inner.data.extend(first.data.iter().map(|(k, vs)| {
+            // # Safety (for aliasing):
+            //
+            // We are aliasing every value in the read map, and the oplog has no other
+            // pending operations (by the semantics of JustCloneRHandle). For any of the
+            // values we alias to be dropped, the operation that drops it must first be
+            // enqueued to the oplog, at which point it will _first_ go through
+            // absorb_first, which will remove the alias and leave only one alias left.
+            // Only after that, when that operation eventually goes through absorb_second,
+            // will the alias be dropped, and by that time it is the only value.
```

Comment:
I'd like to also see a note here saying that we know that the map we are extending has the same hasher as we do since the maps were originally created as clones of one another using `new_with_empty`.